fs/osfs.py

```diff
@@ -363,7 +366,9 @@ def setinfo(self, path, info):
                     with convert_os_errors('setinfo', path):
                         os.utime(sys_path, (accessed, modified))
 
-    def _scandir(self, path, namespaces=None):
+
+
+    def _scandir_c(self, path, namespaces=None):
```

Comment:
I'd like to suggest a slightly different way of doing the conditional.

```python
try:
    import scandir
except ImportError:
    scandir = None

class OSFS:
    if scandir:
        def _scandir(self, path, namespaces=None, page=None):
            # with scandir implementation
    else:
        def _scandir(self, path, namespaces=None, page=None):
            # without scandir implementation
```

I think that's a little tidier than having separate `_scandir_c` and `scandir_py` functions. What do you think?